library/std/src/sys_common/lazy_box.rs

```diff
@@ -0,0 +1,66 @@
+#![allow(dead_code)] // Only used on some platforms.
+
+// This is used to wrap pthread {Mutex, Condvar, RwLock} in.
+
+use crate::marker::PhantomData;
+use crate::ops::{Deref, DerefMut};
+use crate::ptr::null_mut;
+use crate::sync::atomic::{
+    AtomicPtr,
+    Ordering::{AcqRel, Acquire},
+};
+
+pub(crate) struct LazyBox<T: LazyInit> {
+    ptr: AtomicPtr<T>,
+    _phantom: PhantomData<T>,
+}
+
+pub(crate) unsafe trait LazyInit {
```

Comment:
I had to make this trait `pub(crate)` instead of `pub`, otherwise rustc will start suggesting to implement `std::sys_common::lazy_box::LazyInit` every time a user tries to call a non-existent `new` function. And due to it being a bound on `LazyBox`, the change from `pub` to `pub(crate)` went a bit viral through `std::sys`.